### PR TITLE
Feature/tcvp 1710

### PR DESCRIPTION
**Pull request by lfdgov**
# Description

This PR includes the following proposed change(s):

- JIRA tickets 1710, 1711, 1712, 1713, 1714, 1715

Email / File History

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Updated and added automated testing in JAVA, C# APIs, ran successfully
tested all updated APIs in swagger
Ran all affected UX components from localhost IDE and manually tested
built and tested all changes from docker-compose environments and tested in Swagger and UX

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/frontend/staff-portal/src/app/components/staff-workbench/ticket-info/ticket-info.component.ts [hunk 1]
```diff
@@ -411,9 +411,9 @@ export class TicketInfoComponent implements OnInit {
     // look in list of statutes
     let statute = this.lookupsService.statutes.filter(x => x.__statuteString == __statuteString) as StatuteView[];
     if (statute && statute.length > 0) {
-      this.tempViolationTicketCount.description = statute[0].description;
-      this.tempViolationTicketCount.actOrRegulationNameCode = statute[0].act; // to do break down legal paragraphing
-      let parts = this.unLegalParagraph(statute[0].section);
+      this.tempViolationTicketCount.description = statute[0].descriptionText;
```
Comment (lfdgov) [inline src/frontend/staff-portal/src/app/components/staff-workbench/ticket-info/ticket-info.component.ts:414]: UX would not compile without these changes

**Review by Sharkww — APPROVED** [review]

**Comment by KDNeufeld** [comment]
I'm curious why this wasn't implemented as database triggers on the necessary tables.  
- After INSERT, copy record to log table
- After UPDATE, copy record to log table (type of change, ie, status change, could be detected and an appropriate comment made).
- Before DELETE, copy record to log table

The API and UI could then just read from the log table.

Just wondering if that would have been easier.

File: src/backend/TrafficCourts/Workflow.Service/Services/EmailSenderService.cs [hunk 2]
```diff
@@ -162,47 +151,74 @@ public async Task SendEmailAsync(SendEmail emailMessage, CancellationToken cance
                 _logger.LogError(scfe, "An error connecting to the the SMTP Server");
                 throw new EmailSendFailedException("An error connecting to the the SMTP Server", scfe);
             }
+            finally
+            {
+                await SaveEmailtoHistory(emailMessage, sentSuccessfully);
```
Comment (KDNeufeld) [inline src/backend/TrafficCourts/Workflow.Service/Services/EmailSenderService.cs:156]: Should this be in it's own try-catch?  If this errors, then the error will propagate to the calling method. The initial email could have been send successfully and this saving to history fail.

File: src/backend/TrafficCourts/Workflow.Service/Services/OracleDataApiService.cs [hunk 3]
```diff
@@ -23,9 +23,26 @@ public async Task<long> CreateDisputeAsync(Dispute dispute)
 
         return await _oracleDataApiClient.SaveDisputeAsync(dispute);
     }
+    public async Task<long> CreateFileHistoryAsync(FileHistory fileHistory)
+    {
+        try
+        {
+            return await _oracleDataApiClient.InsertFileHistoryAsync(fileHistory.TicketNumber, fileHistory);
+        } catch (Exception ex)
```
Comment (KDNeufeld) [inline src/backend/TrafficCourts/Workflow.Service/Services/OracleDataApiService.cs:31]: ex variable is unused.
I think in C# you can just not include it
```suggestion
        } catch (Exception)
```

File: src/backend/TrafficCourts/Workflow.Service/Services/OracleDataApiService.cs [hunk 4]
```diff
@@ -23,9 +23,26 @@ public async Task<long> CreateDisputeAsync(Dispute dispute)
...
+        }
+    }
     public async Task<long> CreateEmailHistoryAsync(EmailHistory emailHistory)
     {
-        return await _oracleDataApiClient.InsertEmailHistoryAsync(emailHistory.TicketNumber, emailHistory);
+        try
+        {
+            return await _oracleDataApiClient.InsertEmailHistoryAsync(emailHistory.TicketNumber, emailHistory);
+        }
+        catch (Exception ex)
```
Comment (KDNeufeld) [inline src/backend/TrafficCourts/Workflow.Service/Services/OracleDataApiService.cs:42]: unused variable warning

**Review by KDNeufeld — APPROVED** [review]

File: src/backend/TrafficCourts/Workflow.Service/Services/EmailSenderService.cs (hunk 2 above)
Comment (lfdgov) [reply src/backend/TrafficCourts/Workflow.Service/Services/EmailSenderService.cs:156]: will do

**Comment by lfdgov** [comment]
> I'm curious why this wasn't implemented as database triggers on the necessary tables.
> 
> * After INSERT, copy record to log table
> * After UPDATE, copy record to log table (type of change, ie, status change, could be detected and an appropriate comment made).
> * Before DELETE, copy record to log table
> 
> The API and UI could then just read from the log table.
> 
> Just wondering if that would have been easier.

Good observation
- only specific situations identified by the business were identified as requiring a message in the file history
- in the future additional entries will be added that are not related to insert/update/delete in the OCCAM schema, for example, adding a document in JUSTIN
- your suggestion may have also resulted in a successful implementation, and could potentially be implemented using something automated in Oracle.  I know that in MS SQL Server you can define 'temporal' tables where all changes are tracked in a history table.

File: src/backend/TrafficCourts/Staff.Service/Services/EmailHistoryService.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    private readonly IOracleDataApiClient _oracleDataApi;
+    private readonly ILogger<EmailHistoryService> _logger;
+    private readonly IBus _bus;
+    private readonly IFilePersistenceService _filePersistenceService;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+
+    public EmailHistoryService(
+        IOracleDataApiClient oracleDataApi,
+        IBus bus,
```
Comment (pbolduc) [inline src/backend/TrafficCourts/Staff.Service/Services/EmailHistoryService.cs:21]: no using the bus in this class

**Review by pbolduc — APPROVED** [review]
I had one comment, but doesnt block approval

File: src/backend/TrafficCourts/Staff.Service/Services/EmailHistoryService.cs (hunk 5 above)
Comment (lfdgov) [reply src/backend/TrafficCourts/Staff.Service/Services/EmailHistoryService.cs]: removed the bus, renamed FileHistoryRecord to SaveFileHistoryRecord which reflects the action happening

**Pull request merged**